crates/egui/src/widgets/plot/mod.rs

```diff
@@ -947,6 +1171,39 @@ impl Plot {
             }
         }
 
+        // --- transform initialized
+
+        // Add legend widgets to plot
+        let bounds = transform.bounds();
+        let x_axis_range = bounds.range_x();
+        let x_steps = {
+            let input = GridInput {
+                bounds: (bounds.min[X_AXIS], bounds.max[X_AXIS]),
+                base_step_size: transform.dvalue_dpos()[X_AXIS] * MIN_LINE_SPACING_IN_POINTS * 2.0,
+            };
+            (grid_spacers[X_AXIS])(input)
+        };
+        let y_axis_range = bounds.range_y();
+        let y_steps = {
+            let input = GridInput {
+                bounds: (bounds.min[Y_AXIS], bounds.max[Y_AXIS]),
+                base_step_size: transform.dvalue_dpos()[Y_AXIS] * MIN_LINE_SPACING_IN_POINTS * 2.0,
+            };
+            (grid_spacers[Y_AXIS])(input)
+        };
+        for mut widget in x_axis_widgets {
+            widget.range = x_axis_range.clone();
+            widget.transform = Some(transform);
+            widget.steps = x_steps.clone();
```

Comment:
This clones a `Vec<GridMark>` for each widget, which is a bit costly. If you wrap it in an `Arc` we can save some CPU time here!